### ISSUE TITLE
How to make it work on MuOS

### ISSUE THREAD

**Issue by Dante777** [retro-m-dev/ROMDownloader-Anbernic-StockOS#2]
Here's how to make app work on the latest MuOS.

Simplify start-up script ROMDownloader.sh to correct pathes:

#!/bin/bash

progdir=/mnt/mmc/MUOS/application/ROMDownloader

cd $progdir

program="python3 main.py"

$program

2. Copy ROMDownloader.sh file and ROMDownloader folder to /mnt/mmc/MUOS/application on your console (using SFTP)

3. Login to MuOS terminal with SSH (putty).

4. Install python package installer and PIP library:

# python3 -m ensurepip --default-pip

# pip3 install pillow

5. Run application from MuOS menu Applications->ROMDownloader

**Comment by d0npian0** [comment]
Hey, thanks for this! Been trying to make it work but it just boots me back to the main menu, is the startup script correct?

**Comment by mortizauge** [comment]
Hey, thanks for this! Been trying to make it work but it just boots me back to the main menu, is the startup script 100% correct? I can't see any log files to know what's making it crash

**Comment by mortizauge** [comment]
I added a log to your script and now I'm getting some information about the error. Apparently it's something to do with the font?

`Traceback (most recent call last):
  File "/mnt/mmc/MUOS/application/ROMDownloader/main.py", line 1, in <module>
    import app
  File "/mnt/mmc/MUOS/application/ROMDownloader/app.py", line 2, in <module>
    import graphic as gr
  File "/mnt/mmc/MUOS/application/ROMDownloader/graphic.py", line 14, in <module>
    fontFile[15] = ImageFont.truetype("/usr/share/fonts/TTF/DejaVuSansMono.ttf", 15)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/PIL/ImageFont.py", line 834, in truetype
    return freetype(font)
           ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/PIL/ImageFont.py", line 831, in freetype
    return FreeTypeFont(font, size, index, encoding, layout_engine)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/PIL/ImageFont.py", line 257, in __init__
    self.font = core.getfont(
                ^^^^^^^^^^^^^
OSError: cannot open resource
`

**Comment by mortizauge** [comment]
Yup that's what it was. I just downloaded DejaVuSansMono and placed it in `/usr/share/fonts/TTF ` and now it's working!

**Comment by massiveinvisible** [comment]
> Yup that's what it was. I just downloaded DejaVuSansMono and placed it in `/usr/share/fonts/TTF` and now it's working!

How do I access the /usr/ directory? I can't see it from Win-SCP

**Comment by mortizauge** [comment]
> How do I access the /usr/ directory? I can't see it from Win-SCP

You need to sftp to port 22 with root / root as user and password

**Comment by massiveinvisible** [comment]
Got it working, thanks! Any idea where it's saving roms and how to make it save to my MUOS ROM folders?

**Comment by mortizauge** [comment]
> Got it working, thanks! Any idea where it's saving roms and how to make it save to my MUOS ROM folders?

Yep, had the same issue. You just have to modify the anbernic.py file, it's in ROMDownloader/anbernic. For example my psp folder is called "Sony PSP" so I have it like this `"PSP": "Sony PSP"`. Hope this helps!

**Comment by mortizauge** [comment]
As an addendum to this, maybe there could be a function that looks for already existing directories using keywords before creating new ones for each system. For example if there's a directory with "Advance" in it, it could autoassign it to the GBA directory. Just a thought. @retro-m-dev

**Comment by retro-m-dev** [comment]
@mortizauge  So MuOS doesn't provide a default location for the roms?

**Comment by NickMellon** [comment]
@retro-m-dev it does not - see https://muos.dev/help/addcontent for examples.

In case anyone has difficulty getting this app to work on muOS, I had to update pip to a newer version (iirc from 24.0 to 24.2)

Open terminal and use ` $ pip install --upgrade pip` to do so

**Comment by retro-m-dev** [comment]
@NickMellon Thank you for the link. So, someone might even have roms from different systems in a single folder, right? It doesn't sound good because @mortizauge 's suggestion won't work in this case

I'll try to create a repo with all the dependencies included, so it should be easier to install.
